pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -649,8 +651,14 @@ func (cm *containerManagerImpl) Start(ctx context.Context, node *v1.Node,
 	return nil
 }
 
-func (cm *containerManagerImpl) GetPluginRegistrationHandler() cache.PluginHandler {
-	return cm.deviceManager.GetWatcherHandler()
+func (cm *containerManagerImpl) GetPluginRegistrationHandlers(kubeClient clientset.Interface, getNode func() (*v1.Node, error)) map[string]cache.PluginHandler {
+	res := map[string]cache.PluginHandler{pluginwatcherapi.DevicePlugin: cm.deviceManager.GetWatcherHandler()}
+
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.DynamicResourceAllocation) {
+		res[pluginwatcherapi.DRAPlugin] = cache.PluginHandler(draplugin.NewRegistrationHandler(kubeClient, getNode))
+	}
```

Comment:
I'm confused here by the fact that we initialize in two very different ways here (struct member initialization vs function call? or is it a cast?), is this correct?

Reply:
In this step I just moved the creation into the container manager without changing how to create those. More unification would be highly beneficial though, I agree. Maybe as a follow up